src/util/graph.h

```diff
@@ -877,9 +908,14 @@ void output_dot_generic(
   const std::function<
     void(const node_index_type &, std::function<void(const node_index_type &)>)>
     &for_each_succ,
-  const std::function<std::string(const node_index_type &)> node_to_string)
+  const std::function<std::string(const node_index_type &)> node_to_string,
+  const std::function<std::string(const node_index_type &)> node_to_pretty)
 {
   for_each_node([&](const node_index_type &i) {
+    const std::string pretty = node_to_pretty(i);
+    if(pretty == "")
+      return;
+    out << pretty << ";\n";
     for_each_succ(i, [&](const node_index_type &n) {
       out << node_to_string(i) << " -> " << node_to_string(n) << '\n';
```

Comment:
I think this will only produce a correct graph when pretty starts with the result of node_to_string. It would be better to enforce it in some way.

Reply:
Thanks, you're right. I've just pushed a cleaner implementation that enforces this.